Sources/MessageContainerSizeCalculator.swift

```diff
@@ -0,0 +1,55 @@
+/*
+ MIT License
+ 
+ Copyright (c) 2017 MessageKit
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import Foundation
+
+class MessageContainerSizeCalculator {
+
+    func messageContainerSizeFor(messageType: MessageType, with layout: MessagesCollectionViewFlowLayout) -> CGSize {
+        
+        let avatarWidth = layout.avatarSizeCalculator.avatarSizeFor(messageType: messageType, with: layout).width
+        let horizontalOffsets = layout.avatarToEdgePadding + layout.avatarToContainerPadding + layout.messageLeftRightPadding
+        let messageContainerWidth = layout.itemWidth - avatarWidth - horizontalOffsets
+        
+        switch messageType.data {
+        case .text(let text):
+            let messageContainerHeight = text.height(considering: messageContainerWidth, font: layout.messageFont)
+            return CGSize(width: messageContainerWidth, height: messageContainerHeight)
+        default:
+            // TODO: Heights for other types of message data
+            fatalError("Currently .text(String) is the only supported message type")
+        }
+    }
+
+}
+
+fileprivate extension String {
+    
+    func height(considering width: CGFloat, font: UIFont) -> CGFloat {
```

Comment:
Just another tiny suggestion: 

```swift 
func height(considering width: CGFloat, and font: UIFont)
```

would produce a calling site reading experience a bit smoother.
 
```swift
height(considering: width, and: UIFont.systemFont...)
```

Also, this is nice for labels with number of lines = 0. If you want to have a label with max number of lines = 2 for example, this won't do it as it will only consider the text. That way I would prefer the sizeThatFits from the label.